File: frontend/app/actions/query_user_history.js
import { queryUserHistory } from '../api/query_user_history_api';

export async function queryUsersHistory(user_id) {
  try {
      const response = await queryUserHistory(user_id);
  
      if (response.error) {
        return {
          error: response.error
        }
      }
  
      return {
        message: 'User history was found successfully',
        result: response.result,
        success: true
      }
    } catch (error) {
      console.error('Error during user history query:', error);
      return {
        errors: 'An error occured during user history query',
        success: false
      }
    }
}

File: backend/routes/query_user_history.py
from flask import request, jsonify
from utils.mysql_connection import connect_to_database
from __main__ import app

@app.route('/api/query_user_history', methods=['POST'])
def query_user_history():
    data = request.json
    user_id = data.get('user_id')

    conn = connect_to_database()
    cursor = conn.cursor()
    try:
        cursor.execute(
            'SELECT prompt_id, user_input FROM history WHERE user_id = %s ORDER BY created_at DESC',
            (user_id,)
        )
        result = cursor.fetchall()
        return jsonify({'message': 'Found user history successfully',
                        'result': result}), 200
    except:
        return jsonify({'error': 'No user history found'}), 404
    finally:
        cursor.close()
        conn.close()